src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordDeconstructor.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedRecordDeconstructor : SynthesizedInstanceMethodSymbol
+    {
+        private readonly ImmutableArray<Symbol> _members;
+
+        public SynthesizedRecordDeconstructor(
+            SourceMemberContainerTypeSymbol containingType,
+            ImmutableArray<ParameterSymbol> ctorParameters,
+            ImmutableArray<Symbol> members)
+        {
+            _members = members;
+            ContainingType = containingType;
+            Parameters = ctorParameters.SelectAsArray(
+                (param, ordinal, _) =>
+                    SynthesizedParameterSymbol.Create(
+                        this,
+                        param.TypeWithAnnotations,
+                        ordinal,
+                        RefKind.Out,
+                        param.Name),
+                arg: (object?)null);
+        }
+
+        public override ImmutableArray<ParameterSymbol> Parameters { get; }
+
+        public override NamedTypeSymbol ContainingType { get; }
+
+        public override string Name => WellKnownMemberNames.DeconstructMethodName;
+
+        public override MethodKind MethodKind => MethodKind.Ordinary;
+
+        public override int Arity => 0;
+
+        public override bool IsExtensionMethod => false;
+
+        internal override bool HasSpecialName => false;
+
+        internal override MethodImplAttributes ImplementationAttributes => MethodImplAttributes.Managed;
+
+        internal override bool HasDeclarativeSecurity => false;
+
+        internal override MarshalPseudoCustomAttributeData? ReturnValueMarshallingInformation => null;
+
+        internal override bool RequiresSecurityObject => false;
+
+        public override bool HidesBaseMethodsByName => false;
+
+        public override bool IsVararg => false;
+
+        public override bool ReturnsVoid => true;
+
+        public override bool IsAsync => false;
+
+        public override RefKind RefKind => RefKind.None;
+
+        public override TypeWithAnnotations ReturnTypeWithAnnotations => TypeWithAnnotations.Create(
+            isNullableEnabled: true,
+            ContainingType.DeclaringCompilation.GetSpecialType(SpecialType.System_Void));
```

Comment:
`isNullableEnabled: true` argument can be dropped. The other overload of `TypeWithAnnotations.Create()` should be sufficient.

Reply:
Done

Reply:
Consider using a read-only auto-property, assigned in the .ctor, to avoid looking up the type each time. (We'll also want that when we report use-site diagnostics: https://github.com/dotnet/roslyn/issues/44915.)

Reply:
Done